language-design.md

```diff
@@ -17,7 +17,20 @@ For each quote, describe what it was about the quote that led you pick it.
+
+***Quote 2***
+>"We should not make the Java programming language a cathedral, but a plain bazaar might be too loose. What we need is more like a shopping mall, where there are not quite as many choices..." [Steele, 1998, 12]
+
+One of the challenges of language design is properly allowing room for users to extend and change the language. In this quote, Steele references and extends a metaphor that identifies the middle ground between no opportunity for user extension and too much wiggle room to provide cohesion in a language. While he talked about this idea at greater length that what we quoted, we felt that this summed up the key points. Essentially, a cathedral represents a language whose extension is inextensible by its user base. It is far too sturdy a structure and is too imposing to allow for creativity and change. Conversely, the bazaar represents the confusion of too little structure. While it has all the mobility and room for change one might desire, it does so at the cost of unity and clarity. As such, a shopping mall may be the happy middle ground. It has distinct structure as a building but allows for interchangeable parts and development.
+
+***Quote 3***
+>"That’s not Greek, it’s Klingon." [Pavlus, 2012]
+
+This quote, while intended as a witty tongue-in-cheek comment, also manages to encapsulate the main takeaway of Pavlus’s article in one fell swoop. Specifically this comment is poking fun at the inaccessibility of popular programming language syntax, such as Java and Perl, to non-tech-savvy individuals. While Greek may be inaccessible to the average English speaker, that doesn’t mean it is inaccessible to humans as a whole. Rather, there are many Greek speakers across the globe ranging from child to scholar. More specifically, as a spoken language, Greek can be easily used by people of any walk of life to communicate. Klingon, on the other hand, is an impractical alien language that, by lore, is not intended for general humans. Rather, while humans can learn Klingon, it is a language not intended for humans as a whole and only learned by a small dedicated geek subculture, the Star Trek fan base. This parallels the truth of programming languages to the dot. The choices made in programming language syntax design, while logical to the geek subculture of programmers, is inaccessible to general public. Most importantly this quote embodies an important consideration that is mostly overlooked by expert language designers: usability. Many programs, such as the simple 10-iteration loop for the article does not require extensive technical knowledge to understand conceptually and yet the syntax does not reflect this. As the field of programming languages moves forward in the future, Pavlus and the interviewed expert Stefik think that this is something that needs to be addressed.
```

Comment:
This is a good in-depth summary for such a short quote. I question its applicability, because while Klingon may have been designed with express intent of being unreadable, that certainly isn't the case for a language like perl, although the author is arguing that it may as well have been. Having coded in perl I feel like it's not that bad of a language except for the somewhat abstruse syntax, but it wasn't difficult to learn with prior knowledge of other programming languages (although I certainly wouldn't want it as a first language to learn). Usability is certainly important, but it's also important to keep in mind who the target audience is (experienced programmers vs novice programmers vs some other group) and tailor usability to their needs.